lesson_7/lesson_7.py

```diff
@@ -0,0 +1,215 @@
+"""
+# Should print:
+# 3x1=3
+# 3x2=6
+# 3x3=9
+# 3x4=12
+# 3x5=15
+def multiplication_table(number):
+    # Initialize the appropriate variable
+    multiplier = 1
+
+    # Complete the while loop condition.
+    while multiplier <= 10:
+        result = number * multiplier
+        # десь тут помила, а може не одна
+        if  result >= 25:
+            # Enter the action to take if the result is greater than 25
+            break
+        print(str(number) + "x" + str(multiplier) + "=" + str(result))
+
+        # Increment the appropriate variable
+        multiplier += 1
+
+multiplication_table(3)
+
+# task 2
+"""  Написати функцію, яка обчислює суму двох чисел.
+"""
+
+def sum_two_digits(d_1, d_2):
+    return d_1 + d_2
+
+digit_1 = 5
+digit_2 = 7
+
+# dodigit_1 = 10
+# dodigit_2 = 4
+
+print('сума двох чисел:', sum_two_digits(digit_1, digit_2))
+# print(sum_two_digits(digit_1, sum_two_digits(dodigit_1, dodigit_2)))
+
+# task 3
+"""  Написати функцію, яка розрахує середнє арифметичне списку чисел.
+"""
+
+def calculate_average(digits_for_average):
+    if not digits_for_average:
+        return 0
+    return sum(digits_for_average) / len(digits_for_average)
+
+averadge_digits_list_1 = []
+averadge_digits_list_2 = [101, 275, 3777, 754, 15, 336, 6859, 935967, 1, 55]
+averadge_digits_list_3 = [10, 20, 30, 40, 50, 60]
+average = calculate_average(averadge_digits_list_3)
+print(f"Середнє арифметичне: {average}")
+
+# task 4
+"""  Написати функцію, яка приймає рядок та повертає його у зворотному порядку.
+"""
+
+def reverse_direction(str_raw):
+    str_reverse = ''
+    str_len = len(str_raw)
+    for i in reversed(range(str_len)):
+        str_reverse = str_reverse + str_raw[i]
+    return str_reverse
+
+str_to_reverse_1 = "refrigerator"
+reverse = reverse_direction(str_to_reverse_1)
+print(reverse)
+
+
+
+# task 5
+"""  Написати функцію, яка приймає список слів та повертає найдовше слово у списку.
+"""
+
+def the_longest_word_f(list_of_words):
+    max_length_word = ''
+    for word in list_of_words:
+        if len(word) >= len(max_length_word):
+            max_length_word = word
+    return max_length_word
+
+list_of_words_example_1 = ('apple', 'banana', 'pineapple', 'coconut')
+
+the_longest_word = the_longest_word_f(list_of_words_example_1)
+print(the_longest_word)
+
+# task 6
+"""  Написати функцію, яка приймає два рядки та повертає індекс першого входження другого рядка
+у перший рядок, якщо другий рядок є підрядком першого рядка, та -1, якщо другий рядок
+не є підрядком першого рядка."""
+
+def find_substring(str1, str2):
+    return str1.find(str2)
+
+str1 = "Hello, world!"
+str2 = "world"
+print(find_substring(str1, str2)) # поверне 7
+
+str1 = "The quick brown fox jumps over the lazy dog"
+str2 = "cat"
+print(find_substring(str1, str2)) # поверне -1
+
+# task 7 was created by myself
+# To write a function(s) which takes the biggest value from each list and then sum them
+
+def sum_two_biggest_values_1(list_of_values_task_7):
```

Comment:
Here the same thing. We have built in function max for searching the max value from the list without need to creation our own solution. Save your time using more clear and flexible solution with less code implementation.